eslint/plugins/audacia-eslint-plugin-angular/lib/rules/data-test-attribute.js

```diff
@@ -163,63 +200,55 @@ module.exports = {
-                    }
-                    else {
-                        const event = getIncludedEvent(currentNode);
-
-                        if (event) {
-                            message = `Elements with ${event} events should include a '${configuration.testAttribute}' attribute`;
-                        }
-                        else {
-                            continue;
+                let loc;
+
+                if (isIncludedElement(element.name)) {
+                    message = `${element.name} elements should include a '${configuration.testAttribute}' attribute`;
+                    // set the location to just include "<tagname"
+                    loc = {
+                        start: element.loc.start,
+                        end: {
+                            line: element.loc.start.line,
+                            column: element.loc.start.column + element.name.length + 1
                         }
                     }
+                }
+                else {
+                    const event = getIncludedEvent(element);
 
-                    const attribute = getDataTestAttribute(currentNode);
-
-                    if (isAttributeValid(attribute)) {
-                        continue;
+                    if (event) {
+                        message = `Elements with ${event.name} events should include a '${configuration.testAttribute}' attribute`;
+                        loc = event.loc;
                     }
+                    else {
+                        return;
+                    }
+                }
 
-                    let fix = null;
+                const attribute = getDataTestAttribute(element);
 
-                    if (configuration.enableFixer) {
-                        fix = generateFixer(currentNode, attribute);
-                    }
+                if (isAttributeValid(attribute)) {
+                    return;
+                }
 
-                    const loc = getLocation(currentNode);
+                let fix = null;
 
-                    // Don't return node on it's own as the angular ESlint can be aggressive and highligh the whole file.
-                    // Return the location instead to highlight only the issue.
-                    context.report({
-                        currentNode,
-                        message,
-                        fix,
-                        loc
-                    });
+                if (configuration.enableFixer) {
+                    fix = generateFixer(element, attribute);
                 }
-            }
+
+                // Don't return node on it's own as the angular ESlint can be aggressive and highligh the whole file.
+                // Return the location instead to highlight only the issue.
+                context.report({
+                    element,
+                    message,
+                    fix,
+                    loc
```

Comment:
looks like this can now be `undefined` - should we check this before reporting it?

Reply:
I think we either set it or return, can't see a path where we don't do one of these things? So shouldn't be undefined by time it gets to the reporting?

I think even if `loc` is undefined, the reporting will fallback to use the location from the `element` which is passed in.